gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSIntegrationTest.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.fs.gcs;
+
+import static com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemConfiguration.DELEGATION_TOKEN_BINDING_CLASS;
+import static com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemConfiguration.PERMISSIONS_TO_REPORT;
+import static com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemTestHelper.createInMemoryGoogleHadoopFileSystem;
+import static com.google.common.truth.Truth.assertThat;
+import static org.apache.hadoop.fs.FileSystemTestHelper.createFile;
+import static org.junit.Assert.*;
+
+import com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase.GcsFileChecksumType;
+import com.google.cloud.hadoop.fs.gcs.auth.TestDelegationTokenBindingImpl;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.DirectoryNotEmptyException;
+import java.util.*;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.util.Progressable;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration tests for GoogleHadoopFS class. */
+@RunWith(JUnit4.class)
+public class GoogleHadoopFSIntegrationTest {
+  private static URI initUri;
+  private GoogleHadoopFileSystem fs;
+  private FsPermission permissions;
+
+  @Before
+  public void before() throws IOException {
+    initUri = new Path("gs://test/").toUri();
+    fs = createInMemoryGoogleHadoopFileSystem();
+    fs.initialize(initUri, loadConfig());
+    permissions = new FsPermission((short) 000);
+  }
+
+  @After
+  public void after() throws IOException {
+    FileSystem.closeAll();
+  }
+
+  @Test
+  public void testInitializationWithUriAndConf_shouldGiveFsStatusWithNotUsedMemory()
+      throws IOException, URISyntaxException {
+    assertThat(new GoogleHadoopFS(initUri, loadConfig()).getFsStatus().getUsed()).isEqualTo(0);
```

Comment:
Use `fs` for initialization and extract `GoogleHadoopFS` to local  variable here, and below.

Reply:
I extracted this but in the first scenario I am checking constructor without fs passed as an argument. Should it be tested as I did?